test: flaky TestAmbient running on OCP

```
  * Helm uninstall executed successfully
  Operator uninstalled
  * Namespace deleted
  << Timeline
  [FAILED] Unexpected response from sleep-b8467fc67-v8gzl pod
  Expected
      <string>: Active Internet connections (only servers)
      Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
      
  to contain substring
      <string>: 15008
  In [It] at: /work/tests/e2e/ambient/ambient_test.go:396 @ 08/04/26 20:14:47.99
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Summarizing 1 Failure:
  [FAIL] Ambient configuration  for supported versions Istio version 1.24.5 when sample apps are deployed in the cluster [It] has the ztunnel proxy sockets configured in the pod network namespace [smoke, ambient]
  /work/tests/e2e/ambient/ambient_test.go:396
Ran 26 of 60 Specs in 148.310 seconds
FAIL! -- 25 Passed | 1 Failed | 0 Pending | 34 Skipped
--- FAIL: TestAmbient (148.32s)
FAIL
```

logs: https://storage.googleapis.com/test-platform-results/logs/branch-ci-openshift-service-mesh-sail-operator-release-3.0-ocp-4.22-e2e-ocp-arm/2084717843949031424/build-log.txt

--- a/tests/e2e/ambient/ambient_test.go
+++ b/tests/e2e/ambient/ambient_test.go
@@ -259,7 +259,13 @@ profile: ambient`
 					})
 
 					It("has the ztunnel proxy sockets configured in the pod network namespace", func(ctx SpecContext) {
-						checkZtunnelPort(sleepPod.Items[0].Name, common.SleepNamespace)
+						// Ztunnel configures the HBONE port (15008) in the pod network namespace
+						// asynchronously after the pod becomes Ready. Retry to avoid flakes on
+						// slow clusters (e.g. OCP ARM) where the socket may not be visible yet.
+						Eventually(func(g Gomega) {
+							checkZtunnelPort(g, sleepPod.Items[0].Name, common.SleepNamespace)
+						}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).Should(Succeed(),
+							"ztunnel should configure port 15008 in the pod network namespace")
 					})
 
 					It("can access the httpbin service from the sleep pod", func(ctx SpecContext) {
@@ -329,9 +335,9 @@ func getEnvVars(container corev1.Container) []corev1.EnvVar {
 	return container.Env
 }
 
-func checkZtunnelPort(podName, srcNamespace string) {
+func checkZtunnelPort(g Gomega, podName, srcNamespace string) {
 	response, err := k.WithNamespace(srcNamespace).Exec(podName, common.SleepContainerName, "netstat -tlpn")
-	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("error validating the proxy sockets in the %q pod", podName))
+	g.Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("error validating the proxy sockets in the %q pod", podName))
 	// Verify that the HBONE mTLS tunnel port (15008) is listed in the output.
-	Expect(response).To(ContainSubstring("15008"), fmt.Sprintf("Unexpected response from %s pod", podName))
+	g.Expect(response).To(ContainSubstring("15008"), fmt.Sprintf("Unexpected response from %s pod", podName))
 }

